app/controllers/pieces_controller.rb

```diff
@@ -1,14 +1,12 @@
 class PiecesController < ApplicationController
   before_action :authenticate_user!
-  def show
-    @game = Game.find(params[:game_id])
-    @selected_piece = Piece.find(params[:id])
-  end
-
   def update
     @piece = Piece.find(params[:id])
+    return render json: { update_attempt: 'invalid move' } if @piece.valid_move?(params[:piece][:row_coordinate].to_i, params[:piece][:column_coordinate].to_i) == false
```

Comment:
Another way to write this.

Instead of 

`if something == false`

you can do

`unless something`